ApiClient.js

```diff
@@ -1,12 +1,13 @@
 var config = require('./config');
-
+import {AsyncStorage} from 'react-native';
 export default class ApiClient {
-	static get(endpoint, headers) {
+	static async get(endpoint, headers) {
 		return fetch(`${config.API_ADDRESS}${endpoint}`, {
 				method: 'GET',
 				headers: {
 					Accept: 'application/json',
 					'Content-Type': 'application/json',
+					 'Authorization': 'Bearer ' + await AsyncStorage.getItem('@Skybunk:token'),
```

Comment:
I don't think we should be doing this in the ApiClient since then it sends the token along with _every_ request, which might not be necessary. It might not matter now but it doesn't feel right to me, since if a request doesn't need it we probably shouldn't be sending it. It could slow down things too since we're waiting or the token on every request, so for requests where we might already have the token saved locally or it doesn't require a token this adds unnecessary overhead.

I would leave the ApiClient alone, and just pass it in in the headers parameter on each individual request.

Reply:
Incorporating it into the apiClient makes the code a lot cleaner. I could change it to be cached in apiClient, and only have to grab the token when starting the app. This would make it faster overall, though I'm not sure if that would have any security issues.

Reply:
Hmm yeah. I'm not super polarised on this, I think I would lean towards adding auth header on individual requests though. It's more symmetric with how we implement it server-side (requiring `verifyToken` on specific endpoints), and I'm personally ok with being explicit even if it means a few more lines of code. Making a util method or something we can call to get a stored token might help with cleanliness and let us optimise performance centrally.

Reply:
What about having the functions look like get(endpoint, headers, authorized, body){} where authorized is a boolean which controls if the token should be user or not? It would still cache the token to improve performance, eliminate some lines of code from the rest of the app, and still explicitly say if the request needs the credentials or not.

Reply:
Sorry @scholvat I should have replied to this earlier - definitely like this architecture, the api method signatures thing can wait if we want to merge this in soon!